### ISSUE TITLE
event-exporter prevents scaledown by cluster-autoscaler

### ISSUE THREAD

**Issue by matti** [GoogleCloudPlatform/k8s-stackdriver#359]
with message `no.scale.down.node.pod.has.local.storage` -- apparently the hostPath ssl certs prevent this.

related: https://github.com/kubernetes/kubernetes/issues/69696

**Comment by acondrat** [comment]
Same issue here. Is there a way to change the deployment spec in order to annotate PODs with `cluster-autoscaler.kubernetes.io/safe-to-evict: "true"`?

It seems to get reset every time we upgrade the cluster.

**Comment by matti** [comment]
I'm currently tainting my nodes with `googleStayAwayWithYourBrokenSuff=yes:NoExecute`

**Comment by guyguy333** [comment]
Is finally a controller the last option to " force auto inject" this annotation on event-exporter and stackdriver-metadata-agent when missing ? I could write one if there is no better option.

**Comment by matti** [comment]
Doesnt this just apply it once...? When the GKE pods are updated, then the annotation is lost?

**Comment by guyguy333** [comment]
As it's a controller, it tracks changes every time Deployment is edited (manually, by GKE, ...). Every time an updated is detected, it will check if annotations is present. If not, it will add it.

**Comment by matti** [comment]
Okay I'm not familiar with kopf, but will it also update if that controller is momentarily not running? The code says "on update" which sounds like it will miss the event if its not running at the time of update.

Anyway good work.

**Comment by guyguy333** [comment]
Good point. Indeed, a better solution would be to add replicas and implement leader election.

**Comment by matti** [comment]
Just put that (well almost) code in while true loop and keep the pod running.

**Comment by rbabyuk-vs** [comment]
what if you add 'PDB' for those apps? like here https://github.com/kubernetes/kubernetes/issues/69696#issuecomment-1018627861

**Comment by rbabyuk-vs** [comment]
I am about to try https://github.com/redhat-cop/resource-locker-operator 
but it looks overhead for me.
neither argocd can do patch nor terraform patch resource does not look convenient to me.

**Comment by ledmonster** [comment]
I have same issue.
- GKE Version: 1.21.6-gke.1500
- apiVersion: autoscaling/v2beta2

**Comment by araminian** [comment]
According to [GKE Release Notes](https://cloud.google.com/kubernetes-engine/docs/release-notes#October_27_2021) , by upgrading to `GKE 1.22` this problem will be solved.

**Comment by red8888** [comment]
I'm on `1.22.10-gke.600` and Im seeing this still: no.scale.down.node.pod.kube.system.unmovable for event exporter

**Comment by pentago** [comment]
does creating a PDB for event-exporter helps?

**Comment by red8888** [comment]
@pentago we should not have to manage event-exporter at all. GKE should be creating it in a non broken state that doesn't require customer intervention

**Comment by alexppg** [comment]
Another solution would be to just be able to disable it and just create it in the good way. But when destroying it or changing it, it's updated...

**Comment by kalhaarsavaj** [comment]
Check here: https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-autoscaler-visibility#:~:text=%22no.scale.down.node.pod.kube,enable%20cluster%20autoscaler%20to%20move%20Pods%20in%20the%20kube%2Dsystem%20namespace.
